### PR TITLE
Make IOU badges clickable

### PR DESCRIPTION
**Pull request by Gonals**
@Julesssss 🐻 

### Fixed Issues

Fixes https://github.com/Expensify/Expensify/issues/165666

### Tests/QA Steps
- Click the `+` icon and select `Request Money` send a request from userA to userB
- Enter an amount and click `next`
- Select an attendee and then click `Request $x`
- Tap the IOU badge icon in the header of the DM 
- It should take you to the IOU report

### Tested On

- [x] Web
- [ ] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### Screenshots
No changes to the UI

### CODE REVIEW

**Comment by Gonals** [comment]
@Julesssss off WIP!

File: src/components/IOUBadge.js [hunk 1]
```diff
@@ -4,12 +4,20 @@ import {Text, View} from 'react-native';
...
 import compose from '../libs/compose';
 import withLocalize, {withLocalizePropTypes} from './withLocalize';
 
 const propTypes = {
     /** IOU Report data object */
     iouReport: PropTypes.shape({
+        /** The report ID of the IOU */
+        reportID: PropTypes.number,
+
+        /** The report ID of the chat associated with theIOU */
```
Comment (Julesssss) [inline src/components/IOUBadge.js:18]: NAB: missing a space between the & IOU

**Review by Julesssss — DISMISSED** [review]
Working nicely 👍

This isn't a blocker, but do you think it would be simple to make the Badge highlight instead of the Chat row when selecting the Badge from the LHN?

**Comment by Gonals** [comment]
> Working nicely 👍
> 
> This isn't a blocker, but do you think it would be simple to make the Badge highlight instead of the Chat row when selecting the Badge from the LHN?

Do you mean on click or on hover?

**Review by nickmurray47 — DISMISSED** [review]
works great

**Comment by Julesssss** [comment]
> Do you mean on click or on hover?

I think on-click makes sense, as it has an action (opening the Details). We should just be able to wrap it in pressable?

**Comment by Gonals** [comment]
Comments addressed! Badge is now cool and pressable!

**Review by Julesssss — APPROVED** [review]
Tests well, thanks for making the style change!

**Comment by Julesssss** [comment]
All yours @nickmurray47

**Review by nickmurray47 — APPROVED** [review]

**Pull request merged**